modules/mapper-extras/src/main/java/org/opensearch/index/mapper/ScaledFloatFieldMapper.java

```diff
@@ -280,17 +279,13 @@ public Query rangeQuery(Object lowerTerm, Object upperTerm, boolean includeLower
             Long lo = null;
             if (lowerTerm != null) {
                 double dValue = scale(lowerTerm);
-                if (includeLower == false) {
-                    dValue = Math.nextUp(dValue);
-                }
+                dValue = includeLower ? Math.nextDown(dValue) : Math.nextUp(dValue);
```

Comment:
If `includeLower`=true, why do we need to set it to `Math.nextDown(dValue)`?

Reply:
This will change the behavior 
```
assertEquals("scaled_float:[-9223372036854775808 TO 9]", getQueryString(scaledFloatQ));
```
```
org.junit.ComparisonFailure: 
Expected :scaled_float:[-9223372036854775808 TO 9]
Actual   :scaled_float:[-9223372036854775808 TO 10]
<Click to see difference>
```

Reply:
https://github.com/opensearch-project/OpenSearch/pull/19188#discussion_r2325542490 this looks like correct behavior `-0.095` indexed as `-9`, now with include lower as false (meaning `>`) and include upper as true. The next integer greater than `-9` is `-8`.

Reply:
Modified the range query with your suggestion to make it simple. I have updated the tests accordingly.